packages/peregrine/lib/talons/CartPage/useCartPage.js

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react';
+
+import { useAppContext } from '../../context/app';
+import { useUserContext } from '../../context/user';
+
+export const useCartPage = () => {
+    const [, appApi] = useAppContext();
+    const { toggleDrawer } = appApi;
+    const [{ isSignedIn }] = useUserContext();
+
+    const handleSignIn = useCallback(() => {
+        // TODO: set navigation state to "SIGN_IN". useNavigation:showSignIn doesn't work.
```

Comment:
Right - the `showSignIn` callback and menu state are local to the navigation component. We would have to hoist the navigation state to a context similar to app. So we could, like we do with `app.toggleDrawer('nav')`, do something like `navigation.toggleView('signIn')`. I'm OK with this just opening the left drawer for now.